### PR TITLE
[improvement](build index)Make build index and clone mutually exclusive and add timeout for index change job

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

Currently the index change job and clone task can be executed at the same time. If the clone task gets stuck at this point, it will cause the index change job to get stuck as well and keep retrying.
To solve this problem, we can refer to alter job and make index change job exclusive with clone task, and introduce the timeout to prevent infinite retries of build index.

Add the following checks and status in FE. 
1. Check if table is stable (build index is not allowed when clone is in progress)
1.1. Tablet is HEALTHY.
1.2. Whether the tablet is included in the Tablet scheduler, if so, it means the current tablet is doing clone.
2. When creating the index change job, set the timeout at the same time.

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run cloud_p1

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/alter/IndexChangeJob.java [hunk 1]
```diff
@@ -307,10 +353,36 @@ protected void runWaitingTxnJob() throws AlterCancelException {
...
+            for (AgentTask task : tasks) {
+                if (task.getFailedTimes() > 3) {
+                    LOG.warn("alter inverted index task failed: " + task.getErrorMsg());
+                    List<Long> failedBackends = failedTabletBackends.computeIfAbsent(task.getTabletId(),
+                            k -> Lists.newArrayList());
+                    failedBackends.add(task.getBackendId());
+                    int expectSucceedTaskNum = tbl.getPartitionInfo()
+                            .getReplicaAllocation(task.getPartitionId()).getTotalReplicaNum();
+                    int failedTaskCount = failedBackends.size();
+                    if (expectSucceedTaskNum - failedTaskCount < expectSucceedTaskNum / 2 + 1) {
```
Comment (xiaokang) [inline fe/fe-core/src/main/java/org/apache/doris/alter/IndexChangeJob.java:380]: expectSucceedTaskNum is based on partition. failedTaskCount is based on task. They are not on the same dimension.

**Review by xiaokang — COMMENTED** [review]
pls add testcase

File: fe/fe-core/src/main/java/org/apache/doris/alter/IndexChangeJob.java (hunk 1 above)
Comment (qidaye) [reply fe/fe-core/src/main/java/org/apache/doris/alter/IndexChangeJob.java:380]: failedTaskCount is the same tablet failed count. expectSucceedTaskNum is the tablet's replica number. 
This check is checking the tasks failed on same tablet.

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**